lib/luadevice.c

```diff
@@ -95,11 +95,7 @@ static int luadevice_fop(lua_State *L, luadevice_t *luadev, const char *fop, int
 		goto err;
 	}
 
-	if (lua_getfield(L, -1, fop) != LUA_TFUNCTION) {
-		lua_getfield(L, -2, "name");
-		pr_err("%s: operation isn't defined for /dev/%s\n", fop, lua_tostring(L, -1));
-		goto err;
-	}
+	lunatik_checkfield(L, -1, fop, LUA_TFUNCTION);
```

Comment:
we cannot raise an error here.. we are not running in a pcall.. I would simply revert this particular change..